packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -0,0 +1,877 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { strict as assert } from "assert";
+import { benchmark, BenchmarkType } from "@fluid-tools/benchmark";
+import { Jsonable } from "@fluidframework/datastore-definitions";
+import {
+	createField,
+	EditableField,
+	FieldKinds,
+	getField,
+	isUnwrappedNode,
+	namedTreeSchema,
+	singleTextCursor,
+} from "../../feature-libraries";
+import { brand, unreachableCase } from "../../util";
+import { ITestTreeProvider, TestTreeProvider } from "../utils";
+import { ISharedTree } from "../../shared-tree";
+import {
+	FieldKindIdentifier,
+	fieldSchema,
+	GlobalFieldKey,
+	IForestSubscription,
+	JsonableTree,
+	LocalFieldKey,
+	moveToDetachedField,
+	NamedTreeSchema,
+	rootFieldKey,
+	rootFieldKeySymbol,
+	SchemaData,
+	TransactionResult,
+	TreeSchemaIdentifier,
+	UpPath,
+	ValueSchema,
+} from "../../core";
+
+/**
+ * Wide - Each node is inserted directly 1 layer below the root node, making a wide tree.
+ * Deep - Each node is inserted one layer below the previous node, making a stick-like tree.
+ */
+enum TreeShape {
+	Wide = 0,
+	Deep = 1,
+}
+
+enum TestPrimitives {
+	Number = 0,
+	Float = 1,
+	String = 2,
+	Boolean = 3,
+	Map = 4,
```

Comment:
Maps generally arn't primitives. I think this enum is misnamed. Also I think this enum adds some unnecessary indirection. I think you would be better off with something like:

```
const testSubtrees: readonly Named<JsonableTree>[] = [{name: "Number", { value: 0, type: "Number" }, ...];
```

That can replace the enum and generateTreeData. It also makes it much easier to add or remove test cases. Remember that one of the goals of how you organize your code should be to make modifications easy, so any time you have a list of things that has to be in more than one place (like the enum and generateTreeData), its a good idea to see if you can unify them.

Reply:
I see you also hard coded the size of this, "4" in multiple places. THats another thing that would need updating if youchange the list of cases which can be fixed by moving to the approach I suggested.

If you were going to keep the enum, you should put a constant right next to it with its length rather than hard coding it in multiple places to reduce the risk of someone modifying it failing to update the other places.

Reply:
removed enum, added map called `testSubtrees` made of `JsonableTree`